### PR TITLE
Refactor PyRegion subclasses to take specific proto object in read

### PR DESCRIPTION
**Pull request by chetan51**
Fixes https://github.com/numenta/nupic/issues/2899.

Builds on https://github.com/numenta/nupic/pull/2887.

Depends on https://github.com/numenta/nupic.core/pull/812.

### CODE REVIEW

**Comment by chetan51** [comment]
@scottpurdy

File: src/nupic/algorithms/cla_classifier_factory.py [hunk 1]
```diff
@@ -57,9 +57,7 @@ def read(proto):
     if impl == 'py':
       return CLAClassifier.read(proto.claClassifier)
     elif impl == 'cpp':
-      instance = FastCLAClassifier()
-      instance.read(proto.claClassifier)
-      return instance
+      return FastCLAClassifier().read(proto.claClassifier)
```
Comment (scottpurdy) [inline src/nupic/algorithms/cla_classifier_factory.py]: Should this be:

```
return FastCLAClassifier.read(proto.claClassifier)
```

**Comment by scottpurdy** [comment]
done reviewing

File: src/nupic/algorithms/cla_classifier_factory.py (hunk 1 above)
Comment (chetan51) [inline src/nupic/algorithms/cla_classifier_factory.py]: You're right! Updated.

**Comment by chetan51** [comment]
@scottpurdy Updated.

**Comment by scottpurdy** [comment]
:+1:

**Pull request merged**